lib/salus/scanners/slither.rb

```diff
@@ -0,0 +1,55 @@
+require 'json'
+require 'salus/scanners/base'
+
+# Slither (https://github.com/crytic/slither) is a Python-based static
+# analysis tool for Solidity smart contracts
+
+module Salus::Scanners
+  class Slither < Base
+    include Salus::Formatting
+
+    def should_run?
+      !!@repository.sol_file_present?
+    end
+
+    def run
+      # Most Solidity projects will be Hardhat (https://hardhat.org/) or Truffle (https://trufflesuite.com/),
+      # which both use NPM to manage Solidity dependencies. We should NPM install in these cases to ensure
+      # Slither has the needed dependencies for scanning
+      run_shell("npm install", chdir: @repository.path_to_repo) if @repository.package_json_present?
```

Comment:
Does `npm install` follow the standard success/failure exit codes?

Reply:
Does `npm install` install dependencies to somewhere outside `@repository.path_to_repo`?